File: code_gen.js
import fs from 'fs';
import path from 'path';
import os from 'os';

// Generate 6 random 6-digit numbers
const numbers = Array.from({ length: 6 }, () => {
  return String(Math.floor(Math.random() * 1_000_000)).padStart(6, '0');
});

// Save to numbers.json in the current directory
fs.writeFileSync(
  path.join(process.cwd(), 'numbers.json'),
  JSON.stringify(numbers, null, 2),
);

// Handle --force or -f
if (process.argv.includes('--force') || process.argv.includes('-f')) {
  const oseDir = path.join(os.homedir(), '.ose');
  if (!fs.existsSync(oseDir)) fs.mkdirSync(oseDir, { recursive: true });

  const tokenPath = path.join(oseDir, 'tokens.json');
  fs.writeFileSync(tokenPath, JSON.stringify(numbers, null, 2));
}

File: code_gen.py
import random
import json
import sys
import os

# Generate 6 random 6-digit number strings
numbers = [f"{random.randint(0, 999999):06d}" for _ in range(6)]

# Save to numbers.json (overwrite each run)
with open("numbers.json", "w") as f:
    json.dump(numbers, f)

# Handle --force or -f argument
if "--force" in sys.argv or "-f" in sys.argv:
    # Ensure the .ose directory exists
    os.makedirs(".ose", exist_ok=True)
    with open(".ose/tokens.json", "w") as f:
        json.dump(numbers, f)
